### ISSUE TITLE
"Save selection" saves more than just selected range

### ISSUE THREAD

**Issue by wizofaus** [musescore/MuseScore#23436]
### Issue type

UX/Interaction bug (incorrect behaviour)

### Bug description

Save selection when a few measures in a particular staff for a particular instrument are selected saves notes from other staves in the same instrument

### Steps to reproduce

1. Create a new grand staff (piano) score
2. Add notes to both staves
3. Select 1 measure, just the top staff
4. Use "save selection" command
5. Open the file created by #4
6. Expected: only 1 staff and notes that were in it. Actual: 2 staves with notes from the other unselected staff.

### MuseScore Version

4.3.1

### Regression

I don't know

### Operating system

Windows 11

### Additional context

Actually wanted "Export to selection", not sure if it's worth putting in a feature request for that.

**Comment by wizofaus** [comment]
Wow, every man and his dog has offered to work on that, yet there's not a single PR...

**Comment by Jojo-Schmitz** [comment]
Isn't this a duplicate of #19091 ?

**Comment by wizofaus** [comment]
No, that's just related to the preview.  This is just a problem with instruments spanning multiple staves, that it saves all staves instead of just the one selected.
